tests/sub_port_interfaces/test_sub_port_interfaces.py

```diff
@@ -66,3 +74,156 @@ def test_packet_routed_with_invalid_vlan(self, duthost, ptfadapter, apply_config
+
+            for next_sub_port, next_value in sub_ports.items():
+                if next_sub_port != sub_port:
+                    generate_and_verify_traffic(duthost=duthost,
+                                                ptfadapter=ptfadapter,
+                                                src_port=next_value['neighbor_port'],
+                                                ip_src=next_value['neighbor_ip'],
+                                                dst_port=next_sub_port,
+                                                ip_dst=next_value['ip'],
+                                                pkt_action='fwd')
+
+            startup_port(duthost, sub_port)
+            generate_and_verify_traffic(duthost=duthost,
+                                        ptfadapter=ptfadapter,
+                                        src_port=value['neighbor_port'],
+                                        ip_src=value['neighbor_ip'],
+                                        dst_port=sub_port,
+                                        ip_dst=value['ip'],
+                                        pkt_action='fwd')
+
+
+    def test_max_numbers_of_sub_ports(self, duthost, ptfadapter, apply_config_on_the_dut, apply_config_on_the_ptf):
+        """
+        Validates that 256 sub-ports can be created per port or LAG
+
+        Test steps:
+            1.) Setup configuration of 256 sub-ports on the DUT.
+            2.) Setup configuration of 256 sub-ports on the PTF.
+            3.) Create ICMP packet.
+            4.) Send ICMP request packet from PTF to DUT.
+            5.) Verify that DUT sends ICMP reply packet to PTF.
+            6.) Clear configuration of sub-ports on the DUT.
+            7.) Clear configuration of sub-ports on the PTF.
+
+        Pass Criteria: PTF gets ICMP reply packet from DUT.
+
+        Note:
+            The running of the test case takes about 80 minutes.
+        """
+        sub_ports_new = dict()
+        sub_ports = apply_config_on_the_dut['sub_ports']
+        sub_ports_new[sub_ports.keys()[0]] = sub_ports[sub_ports.keys()[0]]
+        sub_ports_new[sub_ports.keys()[-1]] = sub_ports[sub_ports.keys()[-1]]
+
+        rand_sub_ports = sub_ports.keys()[random.randint(1, len(sub_ports)-1)]
+        sub_ports_new[rand_sub_ports] = sub_ports[rand_sub_ports]
+
+        for sub_port, value in sub_ports_new.items():
+            generate_and_verify_traffic(duthost=duthost,
+                                        ptfadapter=ptfadapter,
+                                        src_port=value['neighbor_port'],
+                                        ip_src=value['neighbor_ip'],
+                                        dst_port=sub_port,
+                                        ip_dst=value['ip'],
+                                        pkt_action='fwd')
+
+
+    def test_mtu_inherited_from_parent_port(self, duthost, ptfadapter, apply_config_on_the_dut, apply_config_on_the_ptf):
```

Comment:
`ptfadapter` is not used by this case. It can be removed from the argument list.

Reply:
Updated. Thanks!